model/task/db.go

```diff
@@ -214,12 +221,7 @@ var (
 						"$and": []bson.M{
 							{"$eq": []string{"$" + StatusKey, evergreen.TaskUndispatched}},
 							{"$eq": []interface{}{"$" + ActivatedKey, true}},
-							{
-								"$or": []bson.M{
-									{DependsOnKey: 0},
-									{"$ne": []interface{}{"$" + bsonutil.GetDottedKeyName(DependsOnKey, DependencyUnattainableKey), true}},
-								},
-							},
+							{"$eq": []interface{}{isAttainable, true}},
```

Comment:
would it make sense to order these three cases:
1. `TaskUnscheduled`
2. `TaskStatusBlocked`
3. `TaskWillRun`

This would help that we wouldn't need to check `isAttainable` and `ActivatedKey` for this case.

Reply:
If this works, maybe we could get away with using the reduction in place of the comparison to `isAttainable`. This way we wouldn't even need to `addFields`. Something like
```
	"case": bson.M{
		"$and": []bson.M{
			{"$eq": []string{"$" + StatusKey, evergreen.TaskUndispatched}},
			{"$eq": []interface{}{ []interface{}{bson.M{"$reduce": bson.M{
			    "input":        "$" + DependsOnKey,
			    "initialValue": false,
			    "in":           bson.M{"$or": []interface{}{"$$" + bsonutil.GetDottedKeyName("value", DependencyUnattainableKey), "$$" + bsonutil.GetDottedKeyName("this", DependencyUnattainableKey)}},
		        },
		        }, true}}},
		},
	},
	"then": evergreen.TaskStatusBlocked,
```

Reply:
That does make sense. But i think it might be better to be explicit about the requirements for each status as it helps with readability and makes it easier to understand what the code is doing. What do you think?